### ISSUE TITLE
xDscWebService: DefaultAppPool crash. AcceptSelfSignedCertificates should default to $false

### ISSUE THREAD

**Issue by gigi81** [PowerShell/xPSDesiredStateConfiguration#104]
Trying to use the xDscWebService resource, the DefaultAppPool crashes on a 64bit win2012 server with the following message:

The Module DLL 'C:\Windows\SysWOW64\inetsrv\IISSelfSignedCertModule.dll' could not be loaded due to a configuration problem. The current configuration only supports loading images built for a AMD64 processor architecture. The data field contains the error number. To learn more about this issue, including how to troubleshooting this kind of processor architecture mismatch error, see http://go.microsoft.com/fwlink/?LinkId=29349.

Application pool 'DefaultAppPool' is being automatically disabled due to a series of failures in the process(es) serving that application pool.

Checking on the administration.config file, this line was added:
<add name="IISSelfSignedCertModule(32bit)" image="C:\Windows\SysWOW64\inetsrv\IISSelfSignedCertModule.dll" />

This configuration is wrong because it must contain the precondition "bitness32".

Even after removing this configuration, iis was still trying to load that dll, so any advice on where else this is configured would be helpful. For the moment the only way I found to temporary fix it is to enable the 32bit compatibility on the app pool.

I also don't understand why this was installed while I was not using ssl on the pull server.

This is the DSC configuration:

xDscWebService PSDSCPullServer 
{ 
    Ensure                  = "Present" 
    EndpointName            = "PSDSCPullServer" 
    Port                    = 8082
    PhysicalPath            = "$env:SystemDrive\inetpub\PSDSCPullServer" 
    CertificateThumbPrint   = "AllowUnencryptedTraffic"  
    ModulePath              = "$env:PROGRAMFILES\WindowsPowerShell\DscService\Modules" 
    ConfigurationPath       = "$env:PROGRAMFILES\WindowsPowerShell\DscService\Configuration"  
    State                   = "Started" 
    DependsOn               = "[WindowsFeature]DSCServiceFeature"  
}

I'm using PS 5.0.10586.117

Thanks
Luigi

**Comment by gigi81** [comment]
It turned out that the AcceptSelfSignedCertificates default value is set to true.
I think that it should be set to false by default, for at least 3 reasons:
- it's not documented (yet)
- it lowers security standards
- it doesn't work properly (it installs x86 binaries on x64 systems)

**Comment by NarineM** [comment]
@gigi81 
Yes, I saw it too. I am working on updating the resource to enable self signing only when the resource instance sets this property. Thanks for reporting this issue.

**Comment by NarineM** [comment]
@gigi81 
IISSelfSignedCertModule.dll is used in the pull scenarios where the nodes (pull clients) are provisioned with registrationKey. This module is not used for setting up self signed certs on the pull server nor in SSL scenario. I understand that the confusion came from the dll name. Let us leave this issue on to-do list . Thanks for reporting it.

**Comment by bytejunkie** [comment]
I'm getting this same issue. 
I'm not sure that changing the title is right on this issue. the symptom observed is that the App Pool stops/crashes, so by changing the title you've potentially made this harder to search for. 

I could do with some help working out how to change the AcceptSelfSignedCertificates value too.

**Comment by bytejunkie** [comment]
This is set in the config for the web server where you generate the original config for the pull server.

    xDscWebService PSDSCPullServer {
    <snip>
            AcceptSelfSignedCertificates = $false
    }

**Comment by Zuldan** [comment]
Here is a easy temp fix.

```
# To resolve issue with "The Module DLL 'C:\windows\SysWOW64\inetsrv\IISSelfSignedCertModule.dll' could not be loaded due to a configuration problem."
# See https://github.com/PowerShell/xPSDesiredStateConfiguration/issues/104
# https://powershell.org/forums/topic/defaultapppool-crashes-after-pullserver-installation/
xWebAppPool DefaultAppPool
{
    Name                  = 'DefaultAppPool'
    State                 = 'Started'
    enable32BitAppOnWin64 = $True
}
```

**Comment by HaloX69** [comment]
I am experiencing the same issue whereby other web application pools get shut down due to the following error "The Module DLL 'C:\Windows\SysWOW64\inetsrv\IISSelfSignedCertModule.dll' could not be loaded due to a configuration problem. The current configuration only supports loading images built for a x86 processor architecture."

Strange thing is there is no loaded IIS module for that assembly.  I tried removing the assembly and that didn't work.  Setting enable32BitAppOnWin64 on the web applications causes other problems to other custom applications that worked fine before installing DSC 5.1.   This is a serious issue because even removing DSC doesn't resolve the issue so my custom applications are now no longer accessible.  This has been an issue for several months in 5.0 and now 5.1 is released and we still have this lingering.  What to do now?

**Comment by TravisEz13** [comment]
I recommend we set `AcceptSelfSignedCertificates` to $false by default or even remove the option.  I suspect you'll find that self-signed certificates work without this (at least on 5.0 and newer servers.)

**Comment by pngfei** [comment]
I encountered this issue.
By default, AcceptSelfSignedCertificates = $false, to set AcceptSelfSignedCertificates = $true, I deleted the Website of the Pull server in IIS. And run the Pull Server configuration with AcceptSelfSignedCertificates = $true.
In addition to setting AcceptSelfSignedCertificates = $true, actually this is no 'C:\Windows\SysWOW64\inetsrv\IISSelfSignedCertModule.dll', so search the dll in C root, find one in other directory, then I copied it to directory 'C:\Windows\SysWOW64\inetsrv\, then the issue gone

**Comment by TravisEz13** [comment]
@pngfei Why did you want to enable this?
